### PR TITLE
Generate JSON Schema and Markdown doc for `.sourcekit-lsp/config.json`

### PR DESCRIPTION
**Pull request by kateinoigakukun**
Providing a JSON schema for the configuration file should improve the developer experience thanks to better auto-completion and diagnostics support provided by some editors.

Additionally, we have been manually maintaining the configuration file format documentation in `Documentation/Configuration File.md`, but it's easy for the documentation to get out of sync with the actual schema.

This change introduces a new tool, `ConfigSchemaGen`, that generates a JSON schema and a Markdown document for the configuration file based on the Swift type definitions in the `SKOptions` module by using swift-syntax.

## Plan for vscode-swift integration

VSCode exposes a way to feed a JSON Schema definition for an opening file based on file name:
https://code.visualstudio.com/api/references/contribution-points#contributes.jsonValidation

> Contribute a validation schema for a specific type of json file. The url value can be either a local path to a schema file included in the extension or a > remote server URL such as a json schema store.
>
> ```json
> {
>   "contributes": {
>     "jsonValidation": [
>       {
>         "fileMatch": ".jshintrc",
>         "url": "https://json.schemastore.org/jshintrc"
>       }
>     ]
>   }
> }
> ```

Since the schema URL can be a local path and [the configuration field can be modified dynamically](https://github.com/microsoft/vscode/commit/3af611c7590383f34756ab5cbbe063cdd2f8ebe2) by extension, we can specify a schema file shipped in the selected Swift toolchain. Or to simplify packaging flow, we can embed the schema file within the sourcekit-lsp binary and provide a CLI interface to print it, then vscode-swift can write it out in its extension storage and feed the file to VSCode.

### CODE REVIEW

**Comment by kateinoigakukun** [comment]
@swift-ci test

**Comment by kateinoigakukun** [comment]
@swift-ci test

**Review by ahoppen — COMMENTED** [review]
Thank you Yuta. I look forward to `Configuration File.md` not getting out of sync anymore.

File: Utilities/ConfigSchemaGen/Sources/ConfigSchemaGen/ConfigSchemaGen.swift
[the remark this answers is not in the document]
Comment (kateinoigakukun) [reply Utilities/ConfigSchemaGen/Sources/ConfigSchemaGen/ConfigSchemaGen.swift:1]: Added `./sourcekit-lsp-dev-utils verify-config-schema` 👍

**Comment by kateinoigakukun** [comment]
@swift-ci test

**Comment by kateinoigakukun** [comment]
@swift-ci test

File: Utilities/ConfigSchemaGen/Sources/ConfigSchemaGen/ConfigSchemaGen.swift
[the remark this answers is not in the document]
Comment (ahoppen) [reply Utilities/ConfigSchemaGen/Sources/ConfigSchemaGen/ConfigSchemaGen.swift:1]: We need to add it to build-script for now. GitHub Actions isn’t enabled for the sourcekit-lsp repo yet.

**Review by ahoppen — COMMENTED** [review]
Thanks for migrating this to the `SourceKitLSPDevUtils` package. I have a few comments around access levels, otherwise looks great.

**Review by ahoppen — APPROVED** [review]
Thanks!

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by bnbarham** [comment]
Heh, thanks @kateinoigakukun! I noticed the discrepancy in docs in my PR but only fixed it manually - generating much better 🙇‍♂️!

**Pull request merged**